### PR TITLE
DOC: fill in missing docstrings for GroupBy.agg

### PR DESCRIPTION
**Pull request by natmokval**
xref #71

### CODE REVIEW

File: mkdocs.yml [hunk 1]
```diff
@@ -28,6 +28,7 @@ nav:
     - api-reference/dtypes.md
     - api-reference/dependencies.md
     - api-reference/selectors.md
+    - api-reference/group_by.md
```
Comment (DeaMariaLeon) [inline mkdocs.yml:31]: I wonder if we should order these alphabetically.. ?

**Comment by DeaMariaLeon** [comment]
Thank you @natmokval, it's nice! .. but I did have a question so I leave it to @MarcoGorelli ?

File: narwhals/group_by.py [hunk 2]
```diff
@@ -23,6 +23,86 @@ def __init__(self, df: DataFrame, *keys: str | Iterable[str]) -> None:
...
+            We define library agnostic functions:
+
+            >>> def func(df_any):
+            ...     df = nw.from_native(df_any)
+            ...     df = df.group_by("a").agg(nw.col("b").sum()).sort("a")
+            ...     return nw.to_native(df)
+
+            >>> def func_mult_col(df_any):
+            ...     df = nw.from_native(df_any)
+            ...     df = df.group_by(["a", "b"]).agg(nw.sum("c")).sort("a", "b")
```
Comment (MarcoGorelli) [inline narwhals/group_by.py:66]:
```suggestion
            ...     df = df.group_by("a", "b").agg(nw.sum("c")).sort("a", "b")
```

**Review by MarcoGorelli — APPROVED** [review]
nice one, thanks @natmokval !

File: narwhals/group_by.py (hunk 2 above)
Comment (MarcoGorelli) [reply narwhals/group_by.py]: this is the same, just a little shorter

File: mkdocs.yml (hunk 1 above)
Comment (natmokval) [reply mkdocs.yml:31]: thanks for the comment. I sorted alphabetically `API Reference` and submitted PR #245

**Pull request merged**